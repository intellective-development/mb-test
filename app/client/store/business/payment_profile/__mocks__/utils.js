// @flow

export const tokenizeCard = jest.fn();
