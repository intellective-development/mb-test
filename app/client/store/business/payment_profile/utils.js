// @flow

import braintree_client from 'braintree-web/client';

type CardData = {
  number: string,
  expirationMonth: string,
  expirationYear: string,
  cvv: string,
  billingAddress: {
    postalCode: string
  }
}
type TokenizeOptions = {
  braintree_client_token: string
}

const BRAINTREE_ERROR_BODY = {message: 'Invalid Credit Card'};
export const tokenizeCard = (card_data: CardData, {braintree_client_token}: TokenizeOptions): Promise<string> => {
  const tokenize_card_prom = new Promise((resolve, reject) => {
    braintree_client.create(
      { authorization: braintree_client_token },
      (create_error, client_instance) => {
        if (create_error){
          if (global.Raven) global.Raven.captureMessage(create_error);
          reject({error: BRAINTREE_ERROR_BODY});
          return null;
        }
        return client_instance.request({
          endpoint: 'payment_methods/credit_cards',
          method: 'post',
          data: { creditCard: card_data, options: { validate: false } }
        }, (request_error, response) => {
          if (request_error){
            if (global.Raven) global.Raven.captureMessage(request_error);
            reject({error: BRAINTREE_ERROR_BODY});
          } else {
            const nonce = response.creditCards[0].nonce;
            resolve(nonce);
          }
        });
      }
    );
  });

  return tokenize_card_prom;
};
