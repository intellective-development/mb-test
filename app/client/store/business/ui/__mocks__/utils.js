// @flow

export const enterStore = jest.fn();
