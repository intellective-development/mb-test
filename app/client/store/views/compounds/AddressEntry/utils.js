// @flow

import uuid from 'uuid';
import _ from 'lodash';
import type { Address } from 'store/business/address';

// This is a collection of pure functions, essentially abstracted out of the address entry module
// The goal is to have each of these have simple, single responsibility

type SubstringMatch = {length: number, offset: number};
type GoogleAddressTerm = {offset: number, value: string};

// NOTE this is a the type of objects returned from google.maps.places.AutocompleteService
// an is NOT the same as the GooglePlaceAddressComponent type defined below, which defines
// the structure of objects returned from google.maps.places.PlacesService
export type GoogleAddress = {
  description: string,
  id: string,
  matched_substrings: Array<SubstringMatch>,
  place_id: string,
  reference: string,
  structured_formatting: Object, // has main text and substring matches, same for secondary text
  terms: Array<GoogleAddressTerm>,
  types: Array<string> // more of an enum along the lines of 'street_address' | 'geocode' | ...
};

// TODO: should just use legacy_store address models displayShortAddress function
export const addressToString = (address?: Address | GoogleAddress): string => {
  if (!address) return '';
  if (address.description) return address.description; // if google result just return formatted string
  const {address1, address2, city, state, zip_code} = address;
  if (!address1 || !city || !state || !zip_code) return ''; // empty string if not all components present
  let main_text = address1;
  if (address2) main_text = `${main_text}, ${address2}`; // e.g. {mt: '560 Broadway, apt 2', st: 'New York NY 10002'}
  return `${main_text}, ${city} ${state} ${zip_code}`; // e.g. 560 Broadway, New York NY 10002
};

// TODO: write generalized regex special char stripping function
export const matchChunksInString = (full_string: string, chunks: Array<string>) => {
  let test_string = full_string;
  let acc_offset = 0;
  return chunks.filter(c => c.length > 0).reduce((acc, chunk) => { // remove empty strings
    const maybe_match = test_string.match(new RegExp(chunk, 'i'));
    if (!maybe_match) return acc;
    const offset = maybe_match.index;
    const length = maybe_match[0].length;
    const match = {offset: acc_offset + offset, length};
    test_string = test_string.slice(offset + length);
    acc_offset = acc_offset + offset + length;
    return [...acc, match];
  }, []);
};

export const addressToOptionTextProps = ({address1, address2, city, state, zip_code}: Address, input_string: string) => {
  if (!address1 || !city || !state || !zip_code) return ''; // empty string if not all components present
  let main_text = address1;
  if (address2) main_text = `${main_text}, ${address2}`; // e.g. {mt: '560 Broadway, apt 2', st: 'New York NY 10002'}
  const secondary_text = `${city} ${state} ${zip_code}`;
  const input_chunks = input_string ? input_string.split(' ') : [];
  return {
    main_text,
    secondary_text,
    main_text_matches: matchChunksInString(`${main_text},`, input_chunks),
    secondary_text_matches: matchChunksInString(secondary_text, input_chunks)
  };
};

// TODO: remove this in favor of Server deduping?
// just sticking the address1 and zip together so that both are being taken into account
export const uniqAddresses = (addresses: Array<Address>) => (
  _.uniqBy(addresses, (addr) => `${addr.address1}_${addr.zip_code}`)
);

// This is a collection of pure functions, essentially abstracted out of the address entry module
// The goal is to have each of these have simple, single responsibility

const component_names: Array<string> = [
  'street_number',
  'premise',
  'route',
  'sublocality',
  'locality',
  'administrative_area_level_1',
  'postal_code'
];

// type of the object that appears within the address_components array in a google.maps.places.PlacesService response
type GooglePlacesAddressComponent = {long_name: string, short_name: string, types: Array<string>};

export const getStoreableAddress = (response, addresses) => {
  const address_components = response && getAddressComponents(response.address_components);
  const formattedAddress = formatAddressForAPI(address_components, response.geometry);
  return findStoreableAddress(formattedAddress, addresses);
};

export const findStoreableAddress = (formattedAddress, addresses) => {
  const savedAddresses = _.filter(_.values(addresses), ({ id }) => !!id);
  const foundAddresses = _.filter(savedAddresses, _.omit(formattedAddress, ['local_id', 'latitude', 'longitude']));
  // sort those that have id up (descending), pick first, if not fallback to creating new
  return _.head(_.orderBy(foundAddresses, ['id'], ['desc'])) || {local_id: uuid(), ...formattedAddress};
};

// convert googles format to object keyed by the types we want
export const getAddressComponents = (address_components: Array<GooglePlacesAddressComponent>): GoogleAddressObject => {
  const address_obj = {};

  // create an object from the array (based on the type subarray)
  _.forEach(component_names, component_name => {
    address_obj[component_name] = _.find(address_components, component => (
      _.includes(component.types, component_name)
    ));
  });

  address_obj.street_number = address_obj.street_number || address_obj.premise;
  address_obj.city = address_obj.sublocality || address_obj.locality;

  return address_obj;
};

// our object constructed from the parsed google.maps.places.PlacesService result
type GoogleAddressObject = {
  street_number?: GooglePlacesAddressComponent,
  premise?: GooglePlacesAddressComponent,
  route?: GooglePlacesAddressComponent,
  sublocality?: GooglePlacesAddressComponent,
  locality?: GooglePlacesAddressComponent,
  administrative_area_level_1?: GooglePlacesAddressComponent,
  postal_code?: GooglePlacesAddressComponent,
  city?: GooglePlacesAddressComponent
}
type GooglePlacesGeometry = {
  location: {lat: () => string, lng: () => string},
  viewport?: {
    northeast?: {lat: () => string, lng: () => string},
    southwest?: {lat: () => string, lng: () => string}
  }
}

// map to false if no street #
export const validateComponents = (address: GoogleAddressObject) => (address.street_number ? address : false);

export const formatAddressForAPI = (address: GoogleAddressObject, geometry: GooglePlacesGeometry) => {
  const street_number = formatAddressComponent(address.street_number);
  const road = formatAddressComponent(address.route);
  const city = formatAddressComponent(address.city);
  const state = formatAddressComponent(address.administrative_area_level_1, 'short_name');
  const zip_code = formatAddressComponent(address.postal_code);

  // TODO:  Test state formatter
  return {
    address1: `${street_number} ${road}`,
    city: city,
    state: state,
    zip_code: zip_code,
    latitude: geometry.location.lat(),
    longitude: geometry.location.lng()
  };
};

export const formatAddressComponent = (component?: GooglePlacesAddressComponent, length?: string = 'long_name') => (
  component ? component[length] : ''
);

// This creates a memoized function that takes a string1 and string2 and returns true
// if string2's characters appear in order within string1 with 1 char of fuzziness at the end
export const makeFuzzyMatcher = () => {
  const matcherCache = _.memoize(str => ( // interpolate every character with '?' (0 or more matcher)
    new RegExp(`^${str.replace(/./gi, x => (
      /[-[\]/{}()*+?.\\^$|]/i.test(x) ? `\\${x}?` : `${x}?` // escape special characters
    ))}.{1}$`, 'i') // .{1} gives one char of terminal fuzziness, 'i' is case insensitive
  )); // e.g. input '45 Bond' => output new RegExp('^4?5? ?B?o?n?d?.{1}$', 'i')
  return (string1: string, string2: string) => matcherCache(string1).test(string2);
};

export const nonCurrentAddresses = (addresses?: Array<Address>, current_address?: Address): Array<Address> => (
  addresses ? addresses.filter(a => a.local_id !== _.get(current_address, 'local_id')) : []
);
