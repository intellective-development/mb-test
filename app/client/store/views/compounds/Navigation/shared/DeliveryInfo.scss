@import "~client/store/views/style/variables";

.cmNavDeliveryInfo_Container {
  display: flex;
  align-items: center;
}

.cmNavDeliveryInfo_TextTitle {
  @include font-size(12);
  color: $brand-medium-grey;
  letter-spacing: 0.4px;
}

.cmNavDeliveryInfo_AddAddress,
.cmNavDeliveryInfo_Address {
  @include font-size(13);
  @include overflow-ellipsis;

  display: inline-block;
  font-weight: $heavy-font-weight;
  margin-top: 4px;

  max-width: 160px;
}

.cmNavDeliveryInfo_AddAddress {
  color: $brand-red;
}
