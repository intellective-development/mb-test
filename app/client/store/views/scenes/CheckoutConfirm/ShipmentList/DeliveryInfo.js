// @flow

import * as React from 'react';
import { delivery_method_helpers } from 'store/business/delivery_method';

import { MBText } from '../../../elements';

type DeliveryInfoProps = {shipment: Object};
class DeliveryInfo extends React.PureComponent<DeliveryInfoProps> {
  render(){
    const { delivery_method } = this.props.shipment;

    return (
      <div className="csl__delivery__container">
        <div className="csl__delivery__name-container">
          <MBText.Span className="csl__delivery__name">
            {delivery_method_helpers.displayName(delivery_method)}
            <ClosedWarning delivery_method={this.props.shipment.delivery_method} />
          </MBText.Span>
        </div>
        <MBText.Span className="csl__delivery__next-delivery">
          {delivery_method_helpers.formatNextDelivery(delivery_method)}
        </MBText.Span>
      </div>
    );
  }
}

const ClosedWarning = ({delivery_method}) => {
  if (!delivery_method_helpers.isClosed(delivery_method)) return null;

  return <MBText.Span className="csl__delivery__closed">&ensp;Closed</MBText.Span>;
};

export default DeliveryInfo;
