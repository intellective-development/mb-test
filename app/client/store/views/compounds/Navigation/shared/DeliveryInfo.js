// @flow
import * as React from 'react';
import _ from 'lodash';
import { connect } from 'react-redux';
import * as Ent from '@minibar/store-business/src/utils/ent';
import I18n from 'store/localization';
import { address_selectors } from 'store/business/address';
import { supplier_helpers, supplier_selectors } from 'store/business/supplier';
import type { Address } from 'store/business/address';
import type { DeliveryMethod, DeliveryMethodType } from 'store/business/delivery_method';

import { MBLoader, MBText } from '../../../elements';
import { MultiDeliveryMethodIcon } from '../../DeliveryMethodIcon';
import styles from './DeliveryInfo.scss';

type DeliveryInfoProps = {|
  current_address: Address,
  selected_delivery_methods: DeliveryMethod[],
  address_fetching: boolean
|};

const DeliveryInfo = ({current_address, selected_delivery_methods, address_fetching}: DeliveryInfoProps) => {
  let delivery_method_types = selected_delivery_methods.map(dm => dm.type);
  if (_.isEmpty(delivery_method_types)){
    delivery_method_types = ['on_demand'];
  }

  let content;
  if (current_address){
    content = <MBText.Span className={styles.cmNavDeliveryInfo_Address}>{formatAddress(current_address)}</MBText.Span>;
  } else if (address_fetching){
    content = <MBLoader />;
  } else {
    content = <MBText.Span className={styles.cmNavDeliveryInfo_AddAddress}>Add delivery address</MBText.Span>;
  }

  return (
    <div className={styles.cmNavDeliveryInfo_Container}>
      <MultiDeliveryMethodIcon
        width={24}
        height={24}
        delivery_method_types={delivery_method_types} />
      <div>
        <MBText.Span className={styles.cmNavDeliveryInfo_TextTitle}>
          {deliveryMessage(delivery_method_types)}
        </MBText.Span><br />
        {content}
      </div>
    </div>
  );
};


const DeliveryInfoSTP = () => {
  const findAddress = Ent.find('address');
  const findSuppliers = Ent.find('supplier');
  const findDeliveryMethods = Ent.find('delivery_method');

  return (state) => {
    const current_suppliers = findSuppliers(state, supplier_selectors.currentSupplierIds(state));
    const display_suppliers = supplier_helpers.displaySuppliers(current_suppliers);

    const selected_delivery_method_ids = display_suppliers.map(s => supplier_selectors.supplierSelectedDeliveryMethod(state, s.id));
    const selected_delivery_methods = findDeliveryMethods(state, selected_delivery_method_ids);

    return {
      current_address: findAddress(state, address_selectors.currentDeliveryAddressId(state)),
      address_fetching: address_selectors.isFetching(state),
      selected_delivery_methods
    };
  };
};
const DeliveryInfoContainer = connect(DeliveryInfoSTP)(DeliveryInfo);

export default DeliveryInfoContainer;

// utils

const formatAddress = (address: Address) => {
  if (!address.address2) return address.address1;
  return `${address.address1}, ${address.address2}`;
};

const deliveryMessage = (delivery_method_types: DeliveryMethodType[]) => {
  const uniq_types = _.uniq(delivery_method_types);
  const icon_name = _.sortBy(uniq_types).join('__');

  return I18n.t(`ui.nav.delivery_info.${icon_name}`);
};

export const __private__ = {
  deliveryMessage
};
