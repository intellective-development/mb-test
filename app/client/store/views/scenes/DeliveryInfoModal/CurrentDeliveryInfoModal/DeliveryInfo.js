// @flow

import * as React from 'react';
import _ from 'lodash';
import { connect } from 'react-redux';
import * as Ent from '@minibar/store-business/src/utils/ent';
import { delivery_method_helpers } from 'store/business/delivery_method';
import { supplier_selectors, supplier_helpers } from 'store/business/supplier';
import type { Supplier } from 'store/business/supplier';

import AddressSection from './AddressSection';
import SupplierRow from './SupplierRow';
import { MBModal, MBButton } from '../../../elements';

type DeliveryInfoProps = {
  current_suppliers: Array<Supplier>,
  supplier_fetch_loading: boolean,
  deliveryInfoRouteTo: Function,
  hideModal: Function
};
const DeliveryInfo = ({current_suppliers, hideModal, supplier_fetch_loading, deliveryInfoRouteTo}: DeliveryInfoProps) => {
  return (
    <div>
      <MBModal.SectionHeader
        renderRight={() => <MBModal.Close onClick={hideModal} />} >
        Delivery Address
      </MBModal.SectionHeader>
      <AddressSection deliveryInfoRouteTo={deliveryInfoRouteTo} />
      <div className="currdel__done-container">
        <MBButton expand onClick={hideModal}>Done</MBButton>
      </div>
      <SupplierList
        suppliers={current_suppliers}
        supplier_fetch_loading={supplier_fetch_loading}
        deliveryInfoRouteTo={deliveryInfoRouteTo} />
    </div>
  );
};
const DeliveryInfoSTP = () => {
  const findSuppliers = Ent.query(Ent.find('supplier'), Ent.join('delivery_methods'));
  return (state) => {
    const current_supplier_ids = supplier_selectors.currentSupplierIds(state, { ignore_lazy_loaded: true });
    const current_suppliers = supplier_helpers.primarySuppliers(findSuppliers(state, current_supplier_ids));

    return {
      current_suppliers,
      supplier_fetch_loading: supplier_selectors.fetchLoading(state)
    };
  };
};
const DeliveryInfoContainer = connect(DeliveryInfoSTP)(DeliveryInfo);

// TODO: better splitting by type? right now we're just assuming there's only going to be 1 per type
const SupplierList = ({suppliers, supplier_fetch_loading, deliveryInfoRouteTo}) => {
  if (_.isEmpty(suppliers)) return null;
  let supplier_content;

  if (supplier_helpers.hasSplitSupplierTypes(suppliers)){
    const grouped_suppliers = _.groupBy(suppliers, 'type') || {};
    // if we have split types, we render them out as a list
    supplier_content = (
      _.values(grouped_suppliers).map(supplier_group => {
        const sortedGroup = _.sortBy(supplier_group.map(supplier => {
          return supplier ? {...supplier, open: delivery_method_helpers.isClosed(supplier.delivery_methods[0])} : {};
        }), ['open', 'name']);
        return (
          <div>
            { sortedGroup ? <MBModal.SectionHeader top={false}>{sortedGroup[0].type} Stores</MBModal.SectionHeader> : null }
            <div className="currdel__sr-row">
              { sortedGroup.map(supplier => (
                <SupplierRow
                  supplier={supplier}
                  deliveryInfoRouteTo={deliveryInfoRouteTo} />
              ))}
            </div>
          </div>
        );
      })
    );
  } else {
    // otherwise, we just render it out as a single supplier
    const supplier = suppliers[0];
    supplier_content = (
      <div>
        <MBModal.SectionHeader top={false}>Your Local Store</MBModal.SectionHeader>
        <div className="currdel__sr-row">
          <SupplierRow
            supplier={supplier}
            deliveryInfoRouteTo={deliveryInfoRouteTo} />
        </div>
      </div>
    );
  }

  return (
    <div className="currdel__sl__container">
      {supplier_content}
      <SupplierListLoader is_hidden={!supplier_fetch_loading} />
    </div>
  );
};

const SupplierListLoader = ({is_hidden}) => {
  if (is_hidden) return null;

  return <div className="currdel__sl__loading-overlay" />;
};

export default DeliveryInfoContainer;
