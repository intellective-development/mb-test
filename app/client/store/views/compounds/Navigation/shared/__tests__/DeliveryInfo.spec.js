
import React from 'react';
import address_factory from 'store/business/address/__tests__/address.factory';
import delivery_method_factory from 'store/business/delivery_method/__tests__/delivery_method.factory';
import supplier_factory from 'store/business/supplier/__tests__/supplier.factory';
import { LOADING_STATUS } from '@minibar/store-business/src/utils/fetch_status';

import TestProvider from 'store/views/__tests__/utils/TestProvider';
import DeliveryInfo, { __private__ } from '../DeliveryInfo';

const { deliveryMessage } = __private__;

describe('DeliveryInfo', () => {
  it('renders', () => {
    const initial_state = {
      ...supplier_factory.stateify([
        supplier_factory.build('with_delivery_methods')
      ]),
      ...address_factory.stateify(
        address_factory.build()
      )
    };

    expect(render(
      <TestProvider initial_state={initial_state}>
        <DeliveryInfo />
      </TestProvider>
    )).toMatchSnapshot();
  });

  it('renders with different delivery method types', () => {
    const initial_state = {
      ...supplier_factory.stateify([
        supplier_factory.build({delivery_methods: [delivery_method_factory.build('shipped')]}),
        supplier_factory.build({delivery_methods: [delivery_method_factory.build('pickup')]})
      ]),
      ...address_factory.stateify(
        address_factory.build()
      )
    };

    expect(render(
      <TestProvider initial_state={initial_state}>
        <DeliveryInfo />
      </TestProvider>
    )).toMatchSnapshot();
  });

  it('renders with an address missing', () => {
    const initial_state = {
      ...supplier_factory.stateify([
        supplier_factory.build('with_delivery_methods')
      ])
    };

    expect(render(
      <TestProvider initial_state={initial_state}>
        <DeliveryInfo />
      </TestProvider>
    )).toMatchSnapshot();
  });

  it('renders with an address loading', () => {
    const initial_state = {
      ...supplier_factory.stateify([
        supplier_factory.build('with_delivery_methods')
      ]),
      address: {
        fetch_status: LOADING_STATUS
      }
    };

    expect(render(
      <TestProvider initial_state={initial_state}>
        <DeliveryInfo />
      </TestProvider>
    )).toMatchSnapshot();
  });
});

describe('deliveryMessage', () => {
  it('returns a message for on_demand types', () => {
    expect(deliveryMessage(['on_demand'])).toEqual('Delivery to');
  });

  it('returns a message for pickup types', () => {
    expect(deliveryMessage(['pickup'])).toEqual('In-Store pickup near');
  });

  it('returns a message for shipped types', () => {
    expect(deliveryMessage(['shipped'])).toEqual('Shipping to');
  });

  it('returns a message for on_demand and pickup types', () => {
    expect(deliveryMessage(['on_demand', 'pickup'])).toEqual('Delivery and in-store pickup');
  });

  it('returns a message for on_demand and shipped types', () => {
    expect(deliveryMessage(['on_demand', 'shipped'])).toEqual('Delivery and shipping');
  });

  it('returns a message for on_demand pickup and shipped types', () => {
    expect(deliveryMessage(['pickup', 'shipped'])).toEqual('In-store pickup and shipping');
  });

  it('ignores initial ordering', () => {
    expect(deliveryMessage(['shipped', 'pickup'])).toEqual('In-store pickup and shipping');
  });

  it('ignores repeats when the types are homogenous', () => {
    expect(deliveryMessage(['on_demand', 'on_demand'])).toEqual('Delivery to');
  });

  it('ignores repeats when the types are not homogenous', () => {
    expect(deliveryMessage(['on_demand', 'shipped', 'on_demand'])).toEqual('Delivery and shipping');
  });
});
