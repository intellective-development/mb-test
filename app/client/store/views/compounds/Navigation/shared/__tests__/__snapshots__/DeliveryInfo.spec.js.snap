// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeliveryInfo renders 1`] = `
<div>
  <img
    alt="on_demand"
    class="cmDeliveryMethodIcon_Multi"
    height="24"
    src="/assets/components/compounds/delivery_method_icon/on_demand.svg"
    width="24"
  />
  <div>
    <span
      class="elMBText elMBText_ResetSpacing"
    >
      Delivery to
    </span>
    <br />
    <span
      class="elMBText elMBText_ResetSpacing"
    >
       123 Jump Street, Apt. 10
    </span>
  </div>
</div>
`;

exports[`DeliveryInfo renders with an address loading 1`] = `
<div>
  <img
    alt="on_demand"
    class="cmDeliveryMethodIcon_Multi"
    height="24"
    src="/assets/components/compounds/delivery_method_icon/on_demand.svg"
    width="24"
  />
  <div>
    <span
      class="elMBText elMBText_ResetSpacing"
    >
      Delivery to
    </span>
    <br />
    <div>
      <div
        class="MBLoader_Ball MBLoader_Ball__1"
      />
      <div
        class="MBLoader_Ball MBLoader_Ball__2"
      />
      <div
        class="MBLoader_Ball MBLoader_Ball__3"
      />
    </div>
  </div>
</div>
`;

exports[`DeliveryInfo renders with an address missing 1`] = `
<div>
  <img
    alt="on_demand"
    class="cmDeliveryMethodIcon_Multi"
    height="24"
    src="/assets/components/compounds/delivery_method_icon/on_demand.svg"
    width="24"
  />
  <div>
    <span
      class="elMBText elMBText_ResetSpacing"
    >
      Delivery to
    </span>
    <br />
    <span
      class="elMBText elMBText_ResetSpacing"
    >
      Add delivery address
    </span>
  </div>
</div>
`;

exports[`DeliveryInfo renders with different delivery method types 1`] = `
<div>
  <img
    alt="pickup__shipped"
    class="cmDeliveryMethodIcon_Multi"
    height="24"
    src="/assets/components/compounds/delivery_method_icon/pickup__shipped.svg"
    width="24"
  />
  <div>
    <span
      class="elMBText elMBText_ResetSpacing"
    >
      In-store pickup and shipping
    </span>
    <br />
    <span
      class="elMBText elMBText_ResetSpacing"
    >
       123 Jump Street, Apt. 10
    </span>
  </div>
</div>
`;
